components/candy-machine/UpdateCandyMachine.tsx

```diff
@@ -58,8 +65,8 @@ const UpdateCandyMachine: FC<{
             return false
         }
         let cacheData = await cache.text()
-        let cacheDataJson = JSON.parse(cacheData)
-        if (cacheDataJson.candyMachine !== candyMachineAccount) {
+        let cacheDataJson = JSON.parse(cacheData) as ICache
+        if (cacheDataJson.program.candyMachine !== candyMachineAccount) {
```

Comment:
why this update? was it wrong before?

Reply:
Yes. 
![image](https://user-images.githubusercontent.com/43625217/185358622-ee6faf86-2815-411d-aca3-828228eac8f9.png)
The cache file is as in the picture. So i need to access the program property to get the candyMachine. Before it could not get it and always being a falsy condition.